src/main/java/lotto/constant/Prize.java

```diff
@@ -25,7 +27,20 @@ public static Prize valueOf(int matchCount) {
         return Arrays.stream(values())
                 .filter(prize -> prize.matchCount == matchCount)
                 .findFirst()
-                .orElse(null);
+                .orElse(MISS);
+    }
+
+    public static Prize calculateRank(int count, boolean isBonusMatch) {
+        return Arrays.stream(values())
+                .filter(p -> p.matchCount == count)
+                .findFirst()
+                .map(p -> {
+                    if (p == THIRD || p == SECOND) {
+                        return isBonusMatch ? SECOND : THIRD;
+                    }
+                    return p;
+                })
+                .orElse(MISS);
     }
```

Comment:
Prize열거 타입이 하위 타입들에 대한 책임을 모두 같이 지고 있는 것 같아요. 열거타입이 수십개가 더 추가되면서 조건들이 복잡해질수록 해당 함수는 고치기 더 힘들어질 것 같아요. 현재 열거타입 인스턴스는 matchCount와 prizeMoney 두 개의 인스턴스 필드를 가지고 있는데, Predicate 고차함수를 추가해서 추상화 할 수 있지 않을까요? 

```suggestion
    public static Prize calculateRank(int count, boolean isBonusMatch) {
        return Arrays.stream(values())
                .filter(p -> p.matchPredicate.test(count, isBonusMatch))
                .findFirst()
                .orElse(MISS);
    }
```